overlays/dprint.nix

```diff
@@ -0,0 +1,61 @@
+{ dprint
+, jq
+, fetchurl
+, writeText
+, runCommand
+, makeWrapper
+
+, dprintJson
+}:
+
+let
+  dprintConfig = builtins.fromJSON (builtins.readFile dprintJson);
+
+  hashes = {
+    "https://plugins.dprint.dev/typescript-0.85.0.wasm" = "sha256-FMMoXkWL7oTZYCbh108dJtziUKgvvGH6FwK/cglUGLo=";
+  };
+
+  prefetchPlugin = url: fetchurl {
+    inherit url;
+    sha256 = hashes."${url}" or (throw "No known hash for dprint plugin '${url}', please update release.nix");
+  };
+
+  plugins = map prefetchPlugin dprintConfig.plugins;
+
+  simplifiedConfig = builtins.toJSON {
```

Comment:
If this is only used for the `dprintCache`, consider reducing the scope to make it more obvious. It could even use `passAsFile = [ "simplifiedConfig" ]` to make nix produce the file without a separate derivation.